### PR TITLE
Microsoft.Toolkit.Xamarin.Forms namespace rename.

### PR DESCRIPTION
**Pull request by almirvuk**
### Description of Change ###

This pull request is based on bug #154.

Everything from `XamarinCommunityToolkit.*` a namespace is changed to `Microsoft.Toolkit.Xamarin.Forms.*`

After this rename whole solution is building successfully. 

Note: Still need to rename folders, tried that already but it makes all the files removed and added in sense on git. I want to talk with other community members about this question.

### CODE REVIEW

File: XamarinCommunityToolkit.UnitTests/Converters/ItemSelectedEventArgsConverter_Tests.cs [hunk 1]
```diff
@@ -1,11 +1,11 @@
-﻿using XamarinCommunityToolkit.Converters;
+﻿using Microsoft.Toolkit.Xamarin.Forms.Converters;
```
Comment (bartlannoeye) [inline XamarinCommunityToolkit.UnitTests/Converters/ItemSelectedEventArgsConverter_Tests.cs:1]: Do we care about namespace ordering? If so log an issue and I could fix with R#

Comment (almirvuk) [reply XamarinCommunityToolkit.UnitTests/Converters/ItemSelectedEventArgsConverter_Tests.cs:1]: Usually, I have the habit to order them. In my last two PRs, I wanted to order them from longer to shorten ones. 😊 Not sure if there is the policy on that for the entire project. But I am ok with idea to have them ordered. 😊

**Comment by almirvuk** [comment]
@bartlannoeye I will try with GitKraken for rename, currently, I am using the GitHub Desktop client and it registers the removed/added changes for rename of the folder names.

File: XamarinCommunityToolkit.UnitTests/Converters/ItemSelectedEventArgsConverter_Tests.cs (hunk 1 above)
Comment (bartlannoeye) [reply XamarinCommunityToolkit.UnitTests/Converters/ItemSelectedEventArgsConverter_Tests.cs:1]: Typically it's all System namespaces and then other namespaces, ordered alphabetically

**Comment by jfversluis** [comment]
Looks like the fail was because of the renamed csproj file which wasn't changed in the yaml :) fixed that for you.

We also might want to consider renaming the filesystem folders

**Comment by jfversluis** [comment]
image

:(

Do you have time for this @almirvuk ? I was planning on doing a live stream later today anyway to test some stuff. So else I can go through this later today/tomorrow.

**Comment by almirvuk** [comment]
:( I see it, this is not even with the main branch anymore... I can resolve them but later this night around 8-9 PM (EU time)

**Comment by jfversluis** [comment]
Yeah I just merged Steven's Sample app UI. So it's probably just picking the right namespace and making changes in what he did. Nothing too exciting. Let's coordinate around then on who will do it!

**Comment by almirvuk** [comment]
@jfversluis it should be ok now. 😄

**Comment by jfversluis** [comment]
Some unexpected behavior because we use the Xamarin.Forms namespace which conflicts with the regular Xamarin.Forms namespace. We need to choose something different :(

**Comment by jfversluis** [comment]
Whoops :D Not make these changes too often... 

@almirvuk and others, if you can take the NuGet from [here](https://dev.azure.com/xamarin/_apis/resources/Containers/7206868/nuget?itemPath=nuget%2FMicrosoft.Toolkit.Xamarin.Forms.1.0.0-preview1206.nupkg) and see that no namespace collisions happen with regular `Xamarin.Forms` then it's all good for me!

**Comment by AndreiMisiukevich** [comment]
> Whoops :D Not make these changes too often...
> 
> @almirvuk and others, if you can take the NuGet from [here](https://dev.azure.com/xamarin/_apis/resources/Containers/7206868/nuget?itemPath=nuget%2FMicrosoft.Toolkit.Xamarin.Forms.1.0.0-preview1206.nupkg) and see that no namespace collisions happen with regular `Xamarin.Forms` then it's all good for me!

image

**Comment by jfversluis** [comment]
Oh hm interesting. Can you click on the status checks here in GitHub then to azure DevOps and see the artifacts that were produced?

**Comment by AndreiMisiukevich** [comment]
> Oh hm interesting. Can you click on the status checks here in GitHub then to azure DevOps and see the artifacts that were produced?

yes, that works

**Comment by almirvuk** [comment]
https://giphy.com/gifs/hero0fwar-karmawhore-rhyming-g9582DNuQppxC

**Comment by almirvuk** [comment]
image

@jfversluis @AndreiMisiukevich

**Comment by clairernovotny** [comment]
Hey folks -- Microsoft.Toolkit.* is an issue in that the namespace is already used by a different project and it will cause confusion. 

Can you please rename to something else before you release?

**Comment by jfversluis** [comment]
I don't see how that can be. We've gone through the proper channels for this, has been approved and I feel Microsoft.Toolkit is way too generic and it can easily be identified by the platform that comes after it.

**Comment by michael-hawker** [comment]
Hey @jfversluis, I setup a meeting for us Monday, didn't realize there was a big time zone difference when I tried to set something up today. Please feel free to forward any info you have ahead of time about who you cleared this with, as we haven't been involved in any of these discussions.

As was [called out in the original thread already](https://github.com/xamarin/XamarinCommunityToolkit/issues/114#issuecomment-636378119), there could be confusion around if 'Forms' in out existing naming is WinForms or Xamarin Forms, but _it's also about project ownership and communities_.

Outside of what's technical feasible or not, if we both ship multiple things under the namespace `Microsoft.Toolkit.*` and with similar named packages, _folks will start reporting issues for both of our projects on each others repos_.

It looks like [this project already tried to own](https://www.nuget.org/packages/Xamarin.Toolkit.Effects/) `Xamarin.Toolkit` in the past for naming, was there a problem with going back to use that? It seems like a great more direct name.

**Comment by jfversluis** [comment]
Hey @michael-hawker! Sorry for the confusion. I did not see your meeting, sorry for not being clear and/or the timezone confusion. I was meaning to get back to this on Monday as it's about to come up on bedtime here.

I (and we) mean no harm. I'm sure we can get this resolved and I'd hate to go back and forth out in the open like this so let's make sure we have a meeting about this. 

The thing that surprises me the most is that I have been able to reserve this name and prefix. In fact, I notice that "we" have the verified badge while you haven't. So it seems that your registration might not be done properly?

Anyway. Let's set something up and get it resolved. I'm not out to rub anyone the wrong way :)

**Comment by michael-hawker** [comment]
Thanks @jfversluis for some of the background! Not sure if my Teams messages are going through as I see an 'x' in your status, but let me know if the meeting time I setup works for you on Monday. Definitely just trying to make sure we're all on the same page and get up-to-speed on all the project goals. 🙂

The Windows Community Toolkit is supposed to have the package space reserved on NuGet, but we're a .NET Foundation project, so the NuGet status here being also under `Microsoft.*` in general is an odd space with some issues, so there was definitely a hiccup here (as called out in the thread with the NuGet team).

To clarify you're saying 'Xamarin.Toolkit' was tossed as in the future Xamarin will be known as MAUI, eh? So then something like `MAUI.Toolkit` would work but be more confusing now? We have a similar type of issue with WinUI 3 approaching for us too. Anyway, we can follow-up on the call; as you say it'll be quicker that way.

I definitely see opportunities to collaborate. We're doing a lot more with .NET Standard in the Windows Community Toolkit, so it would make sense to contribute those in one space. For instance, we have a new MVVM library and would love to have some Xamarin expertise/feedback in building out a sample.

On the other-side, we're headed down the WinUI 3 Reunion path vs. MAUI, so I don't know how well-aligned those goals would be; lots to talk about, but let's kick things off Monday. Looking forward to chatting.

**Pull request merged**